### ISSUE TITLE
Effect of batch size on training dynamics

### ISSUE THREAD

**Issue by lxuechen** [PrincetonLIPS/SketchGraphs#7]
I'm running the autoconstraint task, and I realized that the default batch size is 2048 for the released training code. I'm reading the paper and it states that a batch size of 8192 is used. 

I'm currently playing with the code using a batch size of 128, so that things fit on my local desktop GPU, and everything runs smoothly so far. But I was wondering whether the original choice of batch size was mainly to speed up training given a fixed number of epochs, or that it affected performance somehow (in the sense that it may affect test log-likelihood/precision/recall).

**Comment by wendazhou** [comment]
The batch size was mainly chosen for performance reasons, 2048 is about as large as we could fit on a single Titan GPU and we trained on 4 GPUs for a total of 8192 batch size. As the learning rate is scaled with the batch size the performance should be very similar although it would probably take you a very long time to train. Hope that helps!

**Comment by lxuechen** [comment]
Makes sense. Thanks!
